Sources/App/Views/PublicPage.swift

```diff
@@ -33,10 +33,36 @@ class PublicPage {
             ),
             .script(
                 .src("/javascripts/main.min.js?\(resourceReloadQueryString())")
-            )
+            ),
+            analyticsHead()
         )
     }
 
+    /// The Google Tag Manager code to be inserted into the <head> element.
+    /// - Returns: A <script> containing the Google Tag Manager template code.
+    final func analyticsHead() -> Node<HTML.HeadContext> {
+        let environment = (try? Environment.detect()) ?? .development
+        return .if(environment == .production,
```

Comment:
There's a question here of whether this should be `== .production` or `!= development`. I went with the former in the end as it's not super important that this gets a test, but we don't *ever* want this to execute in an environment other than production.

For example, I can see us adding a staging environment, and this code is future proof against that.

Reply:
I think this is the right call for now. I think when we add another env we'll want to drive this from a `GOOGLE_ANALYTICS_KEY` env variable instead and pass it in from deployment.